### DISCUSSION TITLE
Gradle-like wrapper

### DISCUSSION THREAD

**Discussion by ennioVisco** [python-poetry/poetry#8242] [Ideas]
One of the amazing features of Gradle as a building tool is that it has a cross-platform `gradlew` script that automatically fetches the right version of gradle, and with some plugins even of the JVM, so that on first runs it sets up the local development requirements and runtime requirements (see https://docs.gradle.org/current/userguide/gradle_wrapper.html).

It would be cool if Poetry was able to reach a similar thing, essentially:

- set up locally the a new venv at the specified python version in the `pyproject` file
- automatically fetch the poetry version compatible with the `pyproject` file
- execute the operation required by the caller (e.g. install)

**Comment by dimbleby** [comment]
duplicate #4988

**Reply by ennioVisco** [reply]
I apologize, I only searched in discussions, thanks!

**Comment by ennioVisco** [comment]
https://github.com/python-poetry/poetry/issues/4988
